src/ChannelWrapper.ts

```diff
@@ -650,7 +663,7 @@ export default class ChannelWrapper extends EventEmitter {
             },
             options
         );
-        consumer.consumerTag = consumerTag;
+        consumer.consumerTag = consumer.options.consumerTag = consumerTag;
```

Comment:
Yes. This is quite implicit, but when `_reconnectConsumer` is called we reuse the previous consumerTag.

This is accomplished due to this line: https://github.com/jwalton/node-amqp-connection-manager/blob/4ede1b201e09e02b4edc5549f8eceb2bf584fb7a/src/ChannelWrapper.ts#L698

Reply:
Why do we want to reuse the same consumerTag if the caller hasn't chosen a specific tag?

Reply:
Who knows if the caller hasn't chosen a specific consumerTag? And once it was specified, we have to assure that the next reconnection will keep same consumerTag (in current implementation it does not keep).

In my usage with this lib, I do specify a consumerTag to help my visual control in RabbitMQ Management, for example.

Reply:
We know the caller hasn't chosen a specific tag if it isn't included in the options-parameter when calling `consume`.

In the current implementation the consumerTag would be the same during reconnects if the caller had included a specific consumerTag. If so, `consumer.options.consumerTag` would already be defined.

On the other hand, if the caller had not chosen a `consumerTag`, it would now always become the same during reconnects, which might be okay. But we would also be mutating the received `options-object` which doesn't feel great.

The current consumerTag is always stored on `consumer.consumerTag`, so we always have it, no matter if it is changing or not.

Reply:
You're right! Keeping the same consumerTag when it's not specified is useless.

I'll do the removal of this piece of code.

Reply:
Sorry if I came across as negative. I actually think this is great 👌

Reply:
I'm sure you brought up it for a simpler solution 😉